### ISSUE TITLE
Bug: AZ Sql server password complexity via terraform

### ISSUE THREAD

**Issue by nemrood** [hashicorp/terraform-provider-azurerm#15791]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform: 0.14.5
hashicorp/azurerm: 2.81.0
hashicorp/random: 2.3.0

### Affected Resource(s)

azurerm_sql_server

### Terraform Configuration Files

```hcl
resource "azurerm_sql_server" "sql-srv" {
  name                                     = "mssqlserver"
  resource_group_name          = module.resource_group.resource_group_name
  location                     = module.resource_group.resource_group_location
  version                      = "12.0"
  administrator_login                  = var.sql_login
  administrator_login_password = "Th1s$a$trongPw23"

  tags = local.tfsettings.tags
}
```

### Debug Output

Error: waiting on create/update future for SQL Server "mssqlserver" (Resource Group "TDP-MC-DEV-RG"): Code="PasswordNotComplex" Message="Password validation failed. The password does not meet policy requirements because it is not complex enough."

### Expected Behaviour

It should have created the sql server. but it says that its not complex enough.

### Actual Behaviour

The error shown above. (gitlab output)

PLEASE NOTE that when i use the same password to create the resource from the azure portal, it works.

I also tried storing the password as an env var on gitlab, and also tried the random_module to create a random string that respects the complexity (i made it sure respects the azure password complexity and of course verified this on the portal), but still getting the same error.

### Steps to Reproduce

init(success), validate(success), plan(success), apply(fail)

**Comment by neil-yechenwei** [comment]
@nemrood , thanks for raising this issue. After tested with latest azurerm provider, I cannot repro it. Could you have a try below tf config that is similar with yours to see if the issue still exists? Thanks.

```
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-sqls-test01"
  location = "West Europe"
}

resource "azurerm_sql_server" "test" {
  name                         = "acctestsqlservertest01"
  resource_group_name          = azurerm_resource_group.test.name
  location                     = azurerm_resource_group.test.location
  version                      = "12.0"
  administrator_login          = "mradministrator"
  administrator_login_password = "Th1s$a$trongPw23"
}
```

![image](https://user-images.githubusercontent.com/19754191/158313755-c281e25e-8851-4438-b557-584b492cd89e.png)

**Comment by dsfrederic** [comment]
Experiencing the same issue as @Marktech1. 

Were you able to solve it?

**Comment by shreyasoti** [comment]
I am also having the same error, please let me know what the solution for will be it?

**Comment by llStrevensll** [comment]
I recieved the same issue in the my mssql_database resource and only for my azure devops pipeline, using my local machine I didn't get the issue

So, after change the login user name, it worked for me  "administrator_login"

try to test with this value in the login name: "4dm1n157r470r"
As a reference that value is in terraform documentation
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_database

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against v2.81.0 of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
